packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -464,14 +465,49 @@ export class DataStores implements IDisposable {
 		requestHeaderData: RuntimeHeaderData,
 	): Promise<FluidDataStoreContext> {
 		const headerData = { ...defaultRuntimeHeaderData, ...requestHeaderData };
+		const request = { url: id };
+
+		this.checkDeleted(id, request, headerData);
 
 		const context = await this.contexts.getBoundOrRemoted(id, headerData.wait);
-		const request = { url: id };
 		if (context === undefined) {
 			// The requested data store does not exits. Throw a 404 response exception.
 			throw responseToException(create404Response(request), request);
 		}
 
+		this.checkTombstoned(context, request, requestHeaderData);
+
+		return context;
+	}
+
+	private checkDeleted(id: string, request: IRequest, requestHeaderData: RuntimeHeaderData) {
```

Comment:
validateNotDeleted? Since it is doing more than checking, it will throw an error if it is deleted.
Similarly, checkTombstoned -> validateNotTombstoned.